.github/workflows/build-ios.yml

```diff
@@ -0,0 +1,186 @@
+          echo -n "$BUILD_TUNNEL_PROVISION_PROFILE_BASE64" | base64 --decode -o $TPP_PATH
+          cp $TPP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+          
+           # Create ExportOptions.plist
+          echo "$EXPORT_OPTIONS" | base64 --decode > "$EXPORT_OPTIONS_PATH"
+
+      - name: Generate FFI bindings
+        run: |
+          make darwin
+          make ffigen
+
+      - name: Generate app.env
+        env:
+          ANDROID_INTERSTITIAL_AD_ID: ${{ secrets.INTERSTITIAL_AD_UNIT_ID }}
+          IOS_INTERSTITIAL_AD_ID: ${{ secrets.INTERSTITIAL_AD_UNIT_ID_IOS }}
+        run: |
+          touch app.env
+          echo "Android_interstitialAd=$ANDROID_INTERSTITIAL_AD_ID" > app.env
+          echo "IOS_interstitialAd=$IOS_INTERSTITIAL_AD_ID" >> app.env
+
+      - name: Get dependencies & Pod install
+        run: |
+          flutter pub get
+          cd ios
+          pod install
+          cd ..
+
+      - name: Build iOS app
+        env:
+          SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
+          SENTRY_ORG: getlantern
+          SENTRY_PROJECT_IOS: lantern-ios
+        run: make ios-release
+
+      - name: Upload application
+        uses: actions/upload-artifact@v3
+        with:
+          name: Lantern.ipa
+          path: ${{ github.workspace }}/build/ios/ipa/Lantern.ipa
+          retention-days: 2
+
+      - name: Compress artifacts using bzip2
+        run: |
+          cd ${{ github.workspace }}/build/ios/ipa/
+          bzip2 -zk Lantern.ipa
+
+      - name: Copy .ipa and .bz2 to root
+        run: |
+          cp ${{ github.workspace }}/build/ios/ipa/Lantern.ipa ${{ github.workspace }}/Lantern.ipa
+          cp ${{ github.workspace }}/build/ios/ipa/Lantern.ipa.bz2 ${{ github.workspace }}/Lantern.ipa.bz2
+
+      - name: Install s3cmd
+        run: pip install s3cmd
+
+      - name: Set s3cmd permissions
+        run: |
+          echo "[default]" > "$HOME/.s3cfg"
+          echo "access_key = ${{ secrets.AWS_ACCESS_KEY }}" >> "$HOME/.s3cfg"
+          echo "secret_key = ${{ secrets.AWS_SECRET_KEY }}" >> "$HOME/.s3cfg"
+
+      - name: Push binaries to s3
+        env:
+          VERSION: "${{ env.version }}"
+          IPA: "${{ env.prefix }}-${{ env.version }}.ipa"
+          update: "lantern_update_ios-${{inputs.version}}.bz2"
+          update_source: Lantern.ipa.bz2
+        run: |
+          mv ${{ env.update_source }} ${{ env.update }}
+          mv Lantern.ipa "$IPA"
+          cp "$IPA" ${{ env.prefix }}.ipa
+          echo ${{ env.version }} > ${{ env.version_file }}
+          shasum -a 256 "$IPA" | cut -d " " -f 1 > "$IPA".sha256
+          shasum -a 256 ${{ env.update }} | cut -d " " -f 1 > ${{ env.update }}.sha256
+          cp "$IPA".sha256 ${{ env.prefix }}.ipa.sha256
+          s3cmd put --acl-public "$IPA" "$IPA".sha256 ${{ env.update }} ${{ env.update }}.sha256 ${{ env.version_file }} ${{ env.prefix }}.ipa.sha256 ${{ env.prefix }}.ipa "s3://lantern"    
```

Comment:
Do we need to upload the IPA to S3 or the lantern-binaries repo? I don't think we'll be able to distribute it ad-hoc without going through the app store because the developer certificate has to include the target device or the app cannot be installed. Plus, we are limited to a maximum of 100 devices

Reply:
We can remove push IPA to s3, but let's keep it in lantern binaries, in case we have to find the debug symbol or go back to the previous release.

Reply:
@jigar-f If we need to "go back to the previous release", won't that already be uploaded to Testflight?

Reply:
What is the point uploading the IPA in the public lantern-binaries repo? Users can't do anything with it.

Reply:
Testflight expires in 40 days, Now that I am thinking, that should be enough time right?

Reply:
Yeah, I think so!